lib/ansible/modules/cloud/vmware/vmware_guest_tools_upgrade.py

```diff
@@ -0,0 +1,184 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: vmware_guest_tools_upgrade
+short_description: Module to upgrade VMTools
+version_added: "2.7"
+description:
+    - "This module upgrades the VMWare Tools on Windows and Linux guests."
```

Comment:
Add note that - "In order to upgrade VMTools please poweron virtual machine before hand - either 'manually' or using module `vmware_guest_powerstate`" or something similar.